lib/usdUfe/ufe/UsdClipboardCommands.cpp

```diff
@@ -480,20 +497,62 @@ void UsdPasteClipboardCommand::execute()
         }
 
         if (!clipboardPrims.empty()) {
-            auto duplicateCmd
-                = UsdUndoDuplicateSelectionCommand::create(clipboardPrims, dstParentItem);
-            duplicateCmd->execute();
-            appendToVector(duplicateCmd->targetItems(), pasteInfo.successfulPastes);
-            auto tmpTargetItems = duplicateCmd->targetItems();
-            _targetItems.insert(_targetItems.end(), tmpTargetItems.begin(), tmpTargetItems.end());
-            _itemsToSelect.insert(
-                _itemsToSelect.end(), tmpTargetItems.begin(), tmpTargetItems.end());
+            UsdSceneItem::Ptr pasteTarget = dstParentItem;
+            if (pasteAsSibling) {
+                auto hier = Ufe::Hierarchy::hierarchy(dstParentItem);
+                auto parentItem = hier ? hier->parent() : nullptr;
+                auto newPasteTarget = downcast(parentItem);
+                if (newPasteTarget) {
+                    pasteTarget = newPasteTarget;
+                } else {
+                    pasteAsSibling = false;
+                }
+            }
+
+            // When pasting as sibling we have changed the paste target to the parent
+            // which means we might have the same parent as one of the other paste targets.
+            // We don't want to paste twice to that same parent.
+            bool skipThisTarget = false;
+            for (const auto& prevPasteInto : _pasteInfos) {
+                if (prevPasteInto.pasteTarget == pasteTarget->path()) {
+                    skipThisTarget = true;
+                    break;
+                }
+            }
+
+            if (!skipThisTarget) {
+                // When pasting as sibling we'll use an extra paste info since
+                // the paste target is different.
+                Ufe::PasteClipboardCommand::PasteInfo extraPasteInfo;
+                if (pasteAsSibling)
+                    extraPasteInfo.pasteTarget = pasteTarget->path();
+
+                auto duplicateCmd
+                    = UsdUndoDuplicateSelectionCommand::create(clipboardPrims, pasteTarget);
+                duplicateCmd->execute();
+                appendToVector(
+                    duplicateCmd->targetItems(),
+                    pasteAsSibling ? extraPasteInfo.successfulPastes : pasteInfo.successfulPastes);
+                auto tmpTargetItems = duplicateCmd->targetItems();
+                _targetItems.insert(
+                    _targetItems.end(), tmpTargetItems.begin(), tmpTargetItems.end());
+                _itemsToSelect.insert(
+                    _itemsToSelect.end(), tmpTargetItems.begin(), tmpTargetItems.end());
+
+                if (pasteAsSibling) {
+                    _pasteInfos.push_back(std::move(extraPasteInfo));
+                }
+            }
         }
 
         // Add the paste info.
         _pasteInfos.push_back(std::move(pasteInfo));
     }
 
+    if (!pasteAsSibling) {
+        _clipboard->setPasteAsSibling();
+    }
```

Comment:
This case is when the user selects an item, does copy, then selects a new item and does paste. We want to continually paste into this newly selected item.